pkg/build/api/types.go

```diff
@@ -966,6 +987,9 @@ type BuildRequest struct {
 	// TriggeredBy describes which triggers started the most recent update to the
 	// buildconfig and contains information about those triggers.
 	TriggeredBy []BuildTriggerCause
+
+	// StrategyOptions contains additional strategy-specific options for the build
+	StrategyOptions *BuildStrategyOptions
```

Comment:
w/o fundamentally changing how start-build works (to have it create a build object instead of a buildrequest object), I don't know how we get there at this point.